README.md

```diff
@@ -6,16 +6,43 @@
 [![Known Vulnerabilities][snyk-image]][snyk-url]
 [![Apache License][license-image]][license-url]
 
-OpenCensus Web is an implementation of OpenCensus, a toolkit for collecting
-application performance and behavior monitoring data. This library currently
-generates traces for the initial page load of a site in the browser. It supports
-sending the trace sampling decision and trace ID from the web server to the
-browser client to enable latency debugging across the stack.
+OpenCensus Web is an implementation of [OpenCensus][opencensus-url], a toolkit for collecting
+application performance and behavior monitoring data. 
+This library instruments browser applications to collect user-side performance data.
 
 The library is in alpha stage and the API is subject to change.
 
 Please join [gitter][gitter-url] for help or feedback on this project.
 
+## Overview
+
+OpenCensus Web instruments web applications generates traces for the initial page load of a site 
+in the browser and the subsequent on-page user interactions after the initial load. Some features 
+that also provides are automatic tracing for *clicks* and *route transitions*, *custom spans*, and browser 
+[Resource Timing API][resource-timing-url] data.
+
+It supports sending the trace sampling decision and trace ID from the web server to the
+browser client to enable latency debugging across the stack.
+
+## Architecture
+
+OpenCensus Web interacts with three application components:
+
+- *Frontend web server*: renders the initial HTML to the browser including the OpenCensus Web library code 
+  and the necessary configuration (e.g. window.ocAgent). This server would typically be instrumented with an 
+  OpenCensus server-side library (Go, Java, etc.) that can attach the server-side spans to the trace, 
+  that is sending the Trace Context Header in the [W3C draft format][trace-context-url].
```

Comment:
Could you add that it is sending the header via a `window.traceparent` variable?